core/source/Array.mint

```diff
@@ -705,4 +705,31 @@ module Array {
       },
       array)
   }
+
+  fun findByAndMap (
+    method : Function(a, Tuple(Bool, b)),
+    array : Array(a)
+  ) : Maybe(b) {
+    `
+    (() => {
+      for (let item of #{array}) {
+        const [found, value]   = #{method}(item)
+
+        if (found) {
+          return #{Maybe::Just(`value`)}
+        }
+      }
+
+      return #{Maybe::Nothing}
+    })()
+    `
+  }
+
+  fun reverseIf (condition : Bool, array : Array(a)) : Array(a) {
```

Comment:
I'd keep the main receiver always as a first argument to allow chaining:

```elixir
[1, 2, 3]
|> reverseIf "foo" == "bar"
```

```suggestion
  fun reverseIf (array : Array(a), condition : Bool) : Array(a) {
```

ditto all other methods in stdlib (could be worth separate PR changing just that).

Reply:
The pipe operator `|>` puts the left side as the **last parameter**. 

Maybe you are confusing it with an other language? :D

Reply:
Oh, I might :O IIRC Elixir passes it as a 1st argument so that's where I could the inspiration from :D